api/dbadapters/presto.ts

```diff
@@ -0,0 +1,228 @@
+    options: {
+      rowLimit?: number;
+      byteLimit?: number;
+    } = { rowLimit: 1000, byteLimit: 1024 * 1024 }
+  ): Promise<IExecutionResult> {
+    const prestoResult = await prestoExecute(this.client, statement);
+    return { rows: prestoResult.data, metadata: {} };
+  }
+
+  public async withClientLock<T>(callback: (client: IDbClient) => Promise<T>) {
+    return await callback(this);
+  }
+
+  public async evaluate(queryOrAction: QueryOrAction, projectConfig?: dataform.ProjectConfig) {
+    const useSingleQueryPerAction =
+      projectConfig?.useSingleQueryPerAction === undefined ||
+      !!projectConfig?.useSingleQueryPerAction;
+    const validationQueries = collectEvaluationQueries(
+      queryOrAction,
+      useSingleQueryPerAction,
+      (query: string) => (!!query ? `explain ${query}` : "")
+    ).map((validationQuery, index) => ({ index, validationQuery }));
+    const validationQueriesWithoutWrappers = collectEvaluationQueries(
+      queryOrAction,
+      useSingleQueryPerAction
+    );
+
+    const queryEvaluations = new Array<dataform.IQueryEvaluation>();
+    for (const { index, validationQuery } of validationQueries) {
+      let evaluationResponse: dataform.IQueryEvaluation = {
+        status: dataform.QueryEvaluation.QueryEvaluationStatus.SUCCESS
+      };
+      try {
+        await prestoExecute(this.client, validationQuery.query);
+      } catch (e) {
+        evaluationResponse = {
+          // TODO: Parse the error with line number.
+          status: dataform.QueryEvaluation.QueryEvaluationStatus.FAILURE,
+          error: e
+        };
+      }
+      queryEvaluations.push(
+        dataform.QueryEvaluation.create({
+          ...evaluationResponse,
+          incremental: validationQuery.incremental,
+          query: validationQueriesWithoutWrappers[index].query
+        })
+      );
+    }
+    return queryEvaluations;
+  }
+
+  public async catalogs(): Promise<string[]> {
+    const result = await prestoExecute(this.client, "show catalogs");
+    const catalogs = flatten(result.data);
+    return catalogs;
+  }
+
+  public async schemas(): Promise<string[]> {
+    let schemas: string[] = [];
+    await Promise.all(
+      (await this.catalogs()).map(async catalog => {
+        const result = await prestoExecute(this.client, `show schemas from ${catalog}`);
+        schemas = schemas.concat(flatten(result.data).map(schema => `${catalog}.${schema}`));
+      })
+    );
+    // TODO: Currently this returns `catalog.schema` rather than just schemas in order to
+    // conform to string return type. Should this be changed? In tables() it is converted to targets.
+    return schemas;
+  }
+
+  public async createSchema(catalog: string, schema: string): Promise<void> {
+    await prestoExecute(this.client, `create schema if not exists ${catalog}.${schema}`);
+  }
+
+  // TODO: This should take parameters to allow for retrieving from a specific catalog/schema.
```

Comment:
while this is true... it's also true for BQ, snowflake, etc. if you want to keep this TODO, I think it should be moved to the interface.